servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -150,29 +150,26 @@ public void onNext(final ServiceDiscovererEvent<ResolvedAddress> event) {
                         event);
                 final List<Host<ResolvedAddress, C>> activeAddresses =
                         activeHostsUpdater.updateAndGet(RoundRobinLoadBalancer.this, currentAddresses -> {
-                            final List<Host<ResolvedAddress, C>> refreshedAddresses =
-                                    new ArrayList<Host<ResolvedAddress, C>>(currentAddresses);
                             final ResolvedAddress addr = requireNonNull(event.address());
-                            for (int i = 0; i < refreshedAddresses.size(); i++) {
-                                Host<ResolvedAddress, C> host = refreshedAddresses.get(i);
-                                if (host.address.equals(addr)) {
-                                    if (event.isAvailable()) {
-                                        LOGGER.debug("Address {} added but it already exists.", addr);
-                                        return currentAddresses;
-                                    } else {
-                                        refreshedAddresses.remove(i);
-                                        host.markInactive();
-                                        return refreshedAddresses;
-                                    }
-                                }
-                            }
+                            final List<Host<ResolvedAddress, C>> refreshedAddresses;
                             if (event.isAvailable()) {
+                                refreshedAddresses = new ArrayList<>(currentAddresses.size() + 1);
+                                refreshedAddresses.addAll(currentAddresses);
                                 refreshedAddresses.add(new Host<>(addr));
-                                return refreshedAddresses;
+                            } else if (currentAddresses.isEmpty()) {
+                                refreshedAddresses = emptyList();
                             } else {
-                                LOGGER.debug("Address {} removed but it does not exist.", addr);
```

Comment:
An equivalent log may still be useful to detect descrepancies of events from SD or an issue with our matching.

Reply:
I would prefer to avoid modifying control flow just for debug logs if possible (e.g. add `boolean` check it after loop). there is a debug statement before this operation for the event and after to prints all addresses which should be sufficient to detect the no-change scenario. I understand not having this log is dropping a targeted log statement, but this was the state for a while so trying to understand more context (e.g. too difficult to detect in practice, etc..).

Reply:
I changed the control flow so that we now only remove the first match ... so the control flow change would be minimal to add this log statement back. however we would need need to add another log statement if the `oldHosts.isEmpty()` to cover both cases ... still not sure the additional log(s) are necessary. wdyt?

Reply:
I didn't see that we already have logs at start and end of this method, so adding this log makes less sense. Disregard my comment here.